### PR TITLE
DATA-2442 - Allow upload training script to registry via CLI 

### PR DESCRIPTION
**Pull request by tahiyasalam**
### Context
We want to be able to use the CLI to upload ML training scripts (via package upload)

### Follow-ups
- Simultaneously upload and submitting training job: https://viam.atlassian.net/browse/DATA-2449
- Updating training script to make it public: https://viam.atlassian.net/browse/DATA-2507
- Frontend for viewing details page on the registry item (currently cannot view): https://viam.atlassian.net/browse/DATA-2450

### Manual testing
`go run cli/viam/main.go training-script upload --org-id=83a4ce23-c0a5-4104-9e72-6148451fb0d7 --path=cli/module.tar.gz --name=test-upload`

### CODE REVIEW

File: cli/app.go [hunk 1]
```diff
@@ -1417,6 +1424,57 @@ Example:
 				},
 			},
 		},
+		{
+			Name:  "training-script",
+			Usage: "manage training scripts for custom ML training",
+			Subcommands: []*cli.Command{
+				{
+					Name:      "upload",
+					Aliases:   []string{"upload"},
```
Comment (etai-shuchatowitz) [inline cli/app.go:1433]: Nit: this alias is the same as the name

File: cli/ml_training.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	metadata := createMetadata(c.Bool(mlTrainingFlagDraft), c.String(mlTrainingFlagType),
+		c.String(mlTrainingFlagFramework))
+	metadataStruct, err := convertMetadataToStruct(metadata)
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.uploadPackage(c.String(generalFlagOrgID),
```
Comment (etai-shuchatowitz) [inline cli/ml_training.go:22]: It doesn't seem like we do any validation at all. Maybe this is out of scope but couldn't I upload any random file this way (and publish in the registry)?

**Comment by etai-shuchatowitz** [comment]
Not sure what I'm doing wrong but when I uploaded stuff, it didn't seem to work (i.e. I couldn't see it in the UI either on the models page nor at the listed url).

File: cli/ml_training.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+func createMetadata(draft bool, modelType, framework string) MLMetadata {
+	return MLMetadata{
+		Draft:     draft,
+		ModelType: findValueOrSetDefault(modelTypes, modelType, string(ModelTypeUnspecified)),
+		Framework: findValueOrSetDefault(modelFrameworks, framework, string(ModelFrameworkUnspecified)),
+	}
+}
+
+func findValueOrSetDefault(arr []string, val, defaultVal string) string {
```
Comment (etai-shuchatowitz) [inline cli/ml_training.go:89]: We should probably tell the user if their submitted type/framework wasn't found and is defaulting to "Unspecified". Right now, a user will think everything was kosher when there were problems (i.e. a typo won't error out and instead will upload as unspecified).

**Comment by etai-shuchatowitz** [comment]
@tahiyasalam thanks for explaining! Still feeling foolish. Was able to see the created modules in the registry but wasn't able to click into them. Was just getting a 400. What am I missing here?

File: cli/app.go (hunk 1 above)
Comment (tahiyasalam) [reply cli/app.go]: Thank you! Removed

File: cli/ml_training.go (hunk 2 above)
Comment (tahiyasalam) [reply cli/ml_training.go:29]: Great point, that is true yes. Validation would be a part of this ticket: https://viam.atlassian.net/browse/DATA-2487. Once the validation in the BE is in place (as part of the package upload implementation), we would want the CLI to also perform those validations before upload.

File: cli/ml_training.go (hunk 3 above)
Comment (tahiyasalam) [reply cli/ml_training.go]: Updated to now return a default if empty and otherwise error if the users' submitted type/framework wasn't found

**Pull request merged**